CDCgov/prime-reportstream
# Cle/6949 testing mode

This PR ensures the output ProcessingId in translate HL7 is correctly set to T or P based on settings.

Test Steps:
1. Run automated tests
OR
1. Set sender to T, anything you run through should result in T
2. Set sender to P, receiver to T, anything run through should result in T
3. Set sender to P, receiver to P, useTestProcessingMode to true, anything run through should result in T
4. Set sender to P, receiver to P, useTestProcessingMode to false, antying run through should result in P

## Changes
- Update test utils to allow specification of receiver translation
- move FHIR bundle in-memory modification into a helper class
- split out functionality to get the translated hl7 message from the bundle
- update router tests to call the fhir helper
- add translator tests for all use cases

## Checklist

### Testing
- [ x Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
